vectordraw/static/js/src/vectordraw.js

```diff
@@ -0,0 +1,633 @@
+            if (vec_state) {
+                this.renderVector(idx, [vec_state.tail, vec_state.tip]);
+            } else {
+                this.removeVector(idx);
+            }
+        }, this);
+        this.settings.points.forEach(function(point, idx) {
+            var point_state = state.points[point.name];
+            if (point_state) {
+                this.renderPoint(idx, point_state);
+            } else {
+                this.removePoint(idx);
+            }
+        }, this);
+        this.board.update();
+    };
+
+    // Logic for checking answers
+
+    var checkHandlerUrl = runtime.handlerUrl(element, 'check_answer');
+
+    var checkXHR;
+
+    function getInput(vectordraw) {
+        var input = vectordraw.getState();
+
+        // Transform the expected_result setting into a list of checks.
+        var checks = [];
+
+        _.each(vectordraw.settings.expected_result, function(answer, name) {
+            var presence_check = {vector: name, check: 'presence'};
+            if ('presence_errmsg' in answer) {
+                presence_check.errmsg = answer.presence_errmsg;
+            }
+            checks.push(presence_check);
+
+            [
+                'tail', 'tail_x', 'tail_y', 'tip', 'tip_x', 'tip_y', 'coords',
+                'length', 'angle', 'segment_angle', 'segment_coords', 'points_on_line'
+            ].forEach(function(prop) {
+                if (prop in answer) {
+                    var check = {vector: name, check: prop, expected: answer[prop]};
+                    if (prop + '_tolerance' in answer) {
+                        check.tolerance = answer[prop + '_tolerance'];
+                    }
+                    if (prop + '_errmsg' in answer) {
+                        check.errmsg = answer[prop + '_errmsg'];
+                    }
+                    checks.push(check);
+                }
+            });
+        });
+
+        input.checks = checks.concat(vectordraw.settings.custom_checks);
+
+        return input;
+    }
+
+    function updateStatus(data) {
+        var correctness = $('.correctness', element),
+            correctClass = 'checkmark-correct fa fa-check',
+            incorrectClass = 'checkmark-incorrect fa fa-times';
+        if (data.result.ok) {
+            correctness.removeClass(incorrectClass);
+            correctness.addClass(correctClass);
+        } else {
+            correctness.removeClass(correctClass);
+            correctness.addClass(incorrectClass);
+        }
+        $('.status-message', element).text(data.result.msg);
+    }
+
+    function checkAnswer(vectordraw) {
+        if (checkXHR) {
+            checkXHR.abort();
+        }
+        var state = getInput(vectordraw);
+        checkXHR = $.post(checkHandlerUrl, JSON.stringify(state))
+            .success(function(response) {
+                updateStatus(response);
+            });
+    }
+
+    // Initialization logic
+
+    $(function ($) {
```

Comment:
I know this was inserted by xblock-sdk, but I don't quite get the point.  The XBlock init function is called after the DOM is ready anyway, so you might just as well insert the code directly.